src/ResourceManager/Insights/Commands.Insights/Diagnostics/SetAzureRmDiagnosticSettingCommand.cs

```diff
@@ -41,6 +41,27 @@ public class SetAzureRmDiagnosticSettingCommand : ManagementCmdletBase
         public string ResourceId { get; set; }
 
         /// <summary>
+        /// Gets or sets the switch for storage parameter of the cmdlet
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "Disable storage")]
+        [ValidateNotNullOrEmpty]
+        public SwitchParameter DisableStorage { get; set; }
+
+        /// <summary>
+        /// Gets or sets the switch for eventhub parameter of the cmdlet
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "Disable service bus")]
+        [ValidateNotNullOrEmpty]
+        public SwitchParameter DisableServiceBus { get; set; }
+
+        /// <summary>
+        /// Gets or sets the switch for workspace parameter of the cmdlet
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "Disable workspace")]
```

Comment:
@vcanaa please remove the `ValueFromPipelineByPropertyName` attribute from the three `SwitchParameter` parameters you have added